tools/src/launcher/Main.hs

```diff
@@ -240,91 +266,101 @@ main = do
 --
 -- * Update (if we are already up-to-date, nothing will happen).
 -- * Launch the node and the wallet.
 -- * If the wallet exits with code 20, then update and restart, else quit.
 clientScenario
-    :: HasConfigurations
-    => Maybe FilePath                      -- ^ Logger config
+    :: Maybe FilePath                      -- ^ Logger config
     -> (FilePath, [Text], Maybe FilePath)  -- ^ Node, its args, node log
     -> (FilePath, [Text])                  -- ^ Wallet, args
     -> (FilePath, [Text], Maybe FilePath, Maybe FilePath)
     -- ^ Updater, args, updater runner, the update .tar
     -> Int                                 -- ^ Node timeout, in seconds
     -> Maybe String                        -- ^ Report server
-    -> IO ()
+    -> M ()
 clientScenario logConf node wallet updater nodeTimeout report = do
     runUpdater updater
     (nodeHandle, nodeAsync, nodeLog) <- spawnNode node
     walletAsync <- async (runWallet wallet)
-    (someAsync, exitCode) <- liftIO $ waitAny [nodeAsync, walletAsync]
+    (someAsync, exitCode) <- waitAny [nodeAsync, walletAsync]
     if | someAsync == nodeAsync -> do
-             TL.putStrLn $ format ("The node has exited with "%shown) exitCode
+             logWarning $ sformat ("The node has exited with "%shown) exitCode
              whenJust report $ \repServ -> do
-                 TL.putStrLn $ format ("Sending logs to "%stext) (toText repServ)
+                 logInfo $ sformat ("Sending logs to "%stext) (toText repServ)
                  reportNodeCrash exitCode logConf repServ nodeLog
-             putText "Waiting for the wallet to die"
+             logInfo "Waiting for the wallet to die"
              void $ wait walletAsync
        | exitCode == ExitFailure 20 -> do
-             putText "The wallet has exited with code 20"
-             TL.putStrLn $ format ("Killing the node in "%int%" seconds") nodeTimeout
+             logNotice "The wallet has exited with code 20"
+             logInfo $ sformat ("Killing the node in "%int%" seconds") nodeTimeout
              sleep (fromIntegral nodeTimeout)
-             putText "Killing the node now"
-             liftIO $ do
-                 Process.terminateProcess nodeHandle
-                 cancel nodeAsync
+             killNode nodeHandle nodeAsync
              clientScenario logConf node wallet updater nodeTimeout report
        | otherwise -> do
-             TL.putStrLn $ format ("The wallet has exited with "%shown) exitCode
+             logWarning $ sformat ("The wallet has exited with "%shown) exitCode
              -- TODO: does the wallet have some kind of log?
-             putText "Killing the node"
-             liftIO $ do
-                 Process.terminateProcess nodeHandle
-                 cancel nodeAsync
+             killNode nodeHandle nodeAsync
+  where
+    killNode nodeHandle nodeAsync = do
+        logInfo "Killing the node"
+        liftIO $ Process.terminateProcess nodeHandle
```

Comment:
I want to have `tryAny` here and logging an exception. One hypothesis is that `terminateProcess` sometimes fails.

Reply:
Okay, I'll add it (I think that `terminateProcess` doesn't throw exceptions but, well, I'm not sure)

Reply:
@gromakovsky done, please rerereview.

Reply:
> I think that terminateProcess doesn't throw exceptions but, well, I'm not sure

I am not sure either, but I know that `TerminateProcess` from Win API can fail and given that `terminateProcess` returns `()`, I expect it to throw an exception in case of failure (`process` seems to be solid library).